### PR TITLE
Updated github actions for 2024, fix PCA issue related to newest ggplot version

### PR DESCRIPTION
**Pull request by SPPearce**
Github actions now updated to the most recent version of BioConductor, and fixed a bug introduced in the newest ggplot2 version. Now passing tests on Github.

### CODE REVIEW

File: R/PCA.R [hunk 1]
```diff
@@ -615,9 +615,12 @@ getShapeScale <- function(plotData, shape, shapePalette, colourScaleType = NULL,
...
 
-  filledShapes <- ifelse(my_scale_shape$scale_name == "manual" & any(my_scale_shape$palette(1) %in% 21:25),
-                         TRUE, FALSE)
-
+  if(any(my_scale_shape$palette(1) %in% 21:25)){
+    filledShapes <- TRUE
+  } else {
+    filledShapes <- FALSE 
+  }
+
```
Comment (SPPearce) [inline R/PCA.R:623]: @Steven-M-Hill , with the most recent version of ggplot2, `scale_name` doesn't appear to be an option of my_scale_shape, and this was evaluating to NULL.

Comment (Steven-M-Hill) [reply R/PCA.R:623]: I'll take a look

Comment (Steven-M-Hill) [reply R/PCA.R:623]: I think this is fine for now. `my_scale_shape` is either a manual scale or, if no shape aesthetic is used, identity scale. At the moment the latter sets the shape to 16 and is hard-coded.
In the future we may want to add in the option to change the default shape when there is no shape aesthetic. The code here would need to be changed at this point.

**Review by Steven-M-Hill — APPROVED** [review]
All looks good

File: R/PCA.R (hunk 1 above)
Comment (SPPearce) [reply R/PCA.R:623]: Sure, we can always change things later on.

**Pull request merged**